compiler/AST/astutil.cpp

```diff
@@ -169,26 +169,34 @@ void compute_call_sites() {
       fn->calledBy = new Vec<CallExpr*>();
   }
 
-  forv_Vec(CallExpr, call, gCallExprs) {
-    if (call->isEmpty() == true) {
+  forv_Vec(FnSymbol, fn, gFnSymbols) {
+    for_SymbolSymExprs(se, fn) {
+      if (CallExpr* call = toCallExpr(se->parentExpr)) {
+        if (call->isEmpty() == true) {
+          assert(0); // not possible
 
-    } else if (isAlive(call) == false) {
+        } else if (isAlive(call) == false) {
+          assert(0); // not possible
 
-    } else if (FnSymbol* fn = call->resolvedFunction()) {
-      fn->calledBy->add(call);
+        } else if (fn == call->resolvedFunction()) {
+          fn->calledBy->add(call);
```

Comment:
The code is correct - it's not doing this for recursive functions only because the outer loop goes over all FnSymbols; the next loop goes over CallExprs mentioning each FnSymbol; and this conditional checks that the CallExpr mentioned FnSymbol by calling it (vs naming it as an argument, say).